### ISSUE TITLE
Node#wrap() throws NullPointerException on standalone elements

### ISSUE THREAD

**Issue by ripply** [jhy/jsoup#593]
When trying to call wrap(String) on a standalone element, wrap accesses the parent without checking if it is null which causes an exception.

Example

```
Document doc = Jsoup.connect('http://www.google.com').get();
// create a new standalone element
Element standaloneElement = new Element(doc.tag(), doc.baseUri())
// wrap accesses parent which is null
standaloneElement.wrap("<div></div>")
```

And the exception

```
java.lang.NullPointerException
    at org.jsoup.nodes.Node.wrap(Node.java:360)
    at org.jsoup.nodes.Element.wrap(Element.java:453)
```

Tested on version 1.8.2

**Comment by jhy** [comment]
Thanks for the report. I can repro it. A simple guard doesn't help, because the document settings fetch expects a document, and the parse fragment gets the context wrong. Needs more work.

**Comment by sheldonshen** [comment]
@jhy ,can you update the document(https://jsoup.org/cookbook/modifying-data/set-html)?
```java
Element span = doc.select("span").first(); // <span>One</span>
span.wrap("<li><a href='http://example.com/'></a></li>");
System.out.println(span.parent());//new add code 
// now: <li><a href="http://example.com"><span>One</span></a></li>
```

**Comment by jhy** [comment]
This was fixed with 77fcaf4dcdc8fd185e2f5d8eb6aa9c425a38bc9b

**Comment by lambdaupb** [comment]
Maybe close  #1394 (heh, Firewire)
